lib/cache/cache.go

```diff
@@ -708,34 +710,48 @@ func New(config Config) (*Cache, error) {
 	}
 	cs.collections = collections
 
+	if config.Unstarted {
```

Comment:
I'm not sure I follow why we need to add this logic now. Could you do a brief explanation?

Reply:
We need this because the only reasonable way to ensure that the cache is "primed" with all the resources that have been potentially just created for the first time is to start it after the auth is (mostly) initialized. We start the auth cache at the point where we used to create the cache, so it's going to be the same behavior.

I'll add an explanation there, if I don't end up reworking some of this to begin with - this is just because I wanted to embed the cache in `auth.Server` (so we needed to specify a cache at creation time), but if I end up reworking it to add it as part of `Services` it might not be needed.

Reply:
Added some more explanations around.